src/dmd/cparse.d

```diff
@@ -3089,22 +3089,15 @@ final class CParser(AST) : Parser!AST
                 pt = t;
                 return true;
             }
-            if (t.value == TOK.colon)
-            {
-                t = peek(t);
-                if (!isConstantExpression(t))
-                    return false;
-                return true;        // unnamed bit-field
-            }
```

Comment:
Checking for bit-fields is not in the grammar for `declaration`, it really belongs in another function `isCStructDeclaration`, but we don't currently have any need to have that just yet.